File: extract.js
const fs = require('fs');
const path = require('path');

const dir = 'src/main/resources/templates';
const vnPattern = /[àáạảãâầấậẩẫăằắặẳẵèéẹẻẽêềếệểễìíịỉĩòóọỏõôồốộổỗơờớợởỡùúụủũưừứựửữỳýỵỷỹđÀÁẠẢÃÂẦẤẬẨẪĂẰẮẶẲẴÈÉẸẺẼÊỀẾỆỂỄÌÍỊỈĨÒÓỌỎÕÔỒỐỘỔỖƠỜỚỢỞỠÙÚỤỦŨƯỪỨỰỬỮỲÝỴỶỸĐ]/;

const extracted = {};

function walk(directory) {
    const files = fs.readdirSync(directory);
    for (const file of files) {
        const filepath = path.join(directory, file);
        if (fs.statSync(filepath).isDirectory()) {
            walk(filepath);
        } else if (filepath.endsWith('.html')) {
            const content = fs.readFileSync(filepath, 'utf8');
            const lines = content.split('\n');
            for (let i = 0; i < lines.length; i++) {
                if (vnPattern.test(lines[i])) {
                    const cleaned = lines[i].trim();
                    if (!extracted[cleaned] && cleaned.length > 0) {
                        extracted[cleaned] = cleaned;
                    }
                }
            }
        }
    }
}

walk(dir);
fs.writeFileSync('vn_strings.json', JSON.stringify(extracted, null, 2), 'utf8');
console.log('Extracted ' + Object.keys(extracted).length + ' unique lines.');

File: extract.py
import os
import re
import json

directory = r'd:\FLIPPED_CLASS\DONG4\FlippedClass_Project\src\main\resources\templates'
vn_pattern = re.compile(r'[àá??ãâ?????a?????èé???ê?????ìí??iòó??õô?????o?????ùú??uu??????ı???dÀÁ??ÃÂ?????A?????ÈÉ???Ê?????ÌÍ??IÒÓ??ÕÔ?????O?????ÙÚ??UU??????İ???Ğ]')

extracted = {}

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith('.html'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                lines = f.readlines()
            
            for i, line in enumerate(lines):
                if vn_pattern.search(line):
                    # Clean up leading/trailing whitespace for dictionary key
                    cleaned = line.strip()
                    if cleaned not in extracted:
                        extracted[cleaned] = cleaned

with open('vn_strings.json', 'w', encoding='utf-8') as f:
    json.dump(extracted, f, ensure_ascii=False, indent=4)
